doc/source/serve/production.md

```diff
@@ -567,3 +557,216 @@ This section summarizes the best practices when deploying to production:
+
+(serve-in-production-kubernetes)=
+
+## Deploying to Production on Kubernetes
+
+The above sections walk through how to deploy a Serve application manually using the Serve CLI.
+However, the best practice when running on Kubernetes is to use the [RayService] custom resource (CR) that's provided as part of [KubeRay] because it automatically handles important production requirements such as health checking, status reporting, failure recovery, and upgrades.
+
+A [RayService] CR encapsulates a multi-node Ray Cluster and a Serve application that will run on top of it into a single Kubernetes manifest.
+Deploying, upgrading, and getting the status of the application can be done using standard `kubectl` commands.
+This section will walk through how to deploy, monitor, and upgrade the [`FruitStand` example](serve-in-production-example) from above on Kubernetes.
+
+### Installing the KubeRay operator
+
+The first step is to install the `KubeRay` operator into your Kubernetes cluster.
+This will create a pod that runs the `KubeRay` controller to manage resources based on the `RayService` CRs you create.
+
+Install the operator using `kubectl apply` and check that the controller pod is running:
+```console
+$ kubectl apply -k "github.com/ray-project/kuberay/ray-operator/config/default"
+$ kubectl get deployments -n ray-system
+NAME           READY   UP-TO-DATE   AVAILABLE   AGE
+ray-operator   1/1     1            1           40s
+
+$ kubectl get pods -n ray-system
+NAME                            READY   STATUS    RESTARTS   AGE
+ray-operator-75dbbf8587-5lrvn   1/1     Running   0          31s
+```
+
+For more details, see the [KubeRay installation guide](https://ray-project.github.io/kuberay/deploy/installation/).
+
+### Deploying a Serve application
+
+Once the KubeRay controller is running, you can manage your Ray Serve application by creating and updating a `RayService` CR.
+`RayService` custom resources consist of the following:
+- a `KubeRay` `RayCluster` config defining the parameters for the cluster that the Serve application will run on.
+- a Ray Serve [config](serve-in-production-config-file) defining the Serve application to run on the cluster.
+
+When the `RayService` is created, the `KubeRay` controller will first create a Ray cluster using the provided configuration.
+Then, once the cluster is running, it will deploy the Serve application to the cluster using the [REST API](serve-in-production-deploying).
+The controller will also create a Kubernetes Service that can be used to route traffic to the Serve application.
+
+Let's see this in action by deploying the same [`FruitStand` example](serve-in-production-example) from above.
+The Serve config for the example is embedded into [this example `RayService` CR](https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/ray_v1alpha1_rayservice.yaml).
```

Comment:
Should we use a specific commit hash here to prevent accidental breakage if the YAML file is updated?

Reply:
Yeah maybe use this branch? https://github.com/ray-project/kuberay/tree/release-0.3

Reply:
good catch done